File: backend/app/schemas/calendar.py
from typing import Optional
from pydantic import BaseModel
from datetime import datetime

class CalendarEventBase(BaseModel):
    title: str
    description: Optional[str] = None
    start_time: datetime
    end_time: datetime
    is_meeting: Optional[bool] = False
    workspace_id: Optional[int] = None
    color: Optional[str] = "#7C3AED"

class CalendarEventCreate(CalendarEventBase):
    pass

class CalendarEventUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    is_meeting: Optional[bool] = None
    color: Optional[str] = None

class CalendarEventResponse(CalendarEventBase):
    id: int
    user_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/calendar.ts
import apiClient from './client'

export interface CalendarEvent {
  id: number
  user_id: number
  workspace_id?: number
  title: string
  description?: string
  start_time: string
  end_time: string
  is_meeting: boolean
  color?: string
  created_at: string
  updated_at: string
}

export interface CreateEventPayload {
  title: string
  description?: string
  start_time: string
  end_time: string
  is_meeting?: boolean
  workspace_id?: number
  color?: string
}

export interface UpdateEventPayload {
  title?: string
  description?: string
  start_time?: string
  end_time?: string
  is_meeting?: boolean
  color?: string
}

export const fetchEvents = async (): Promise<CalendarEvent[]> => {
  const res = await apiClient.get('/calendar/events')
  return res.data
}

export const createEvent = async (data: CreateEventPayload): Promise<CalendarEvent> => {
  const res = await apiClient.post('/calendar/events', data)
  return res.data
}

export const updateEvent = async (id: number, data: UpdateEventPayload): Promise<CalendarEvent> => {
  const res = await apiClient.put(`/calendar/events/${id}`, data)
  return res.data
}

export const deleteEvent = async (id: number): Promise<CalendarEvent> => {
  const res = await apiClient.delete(`/calendar/events/${id}`)
  return res.data
}
